ntbs-service/Services/NotificationService.cs

```diff
@@ -167,8 +168,22 @@ public async Task SubmitNotification(Notification notification)
             await context.SaveChangesAsync();
         }
         
-        public async Task<Notification> GetNotificationWithAllInfoAsync(int? id) {
+        public async Task<Notification> GetNotificationWithAllInfoAsync(int? id)
+        {
             return await repository.GetNotificationWithAllInfoAsync(id);
         }
+
+        public async Task<Notification> CreateLinkedNotificationAsync(Notification notification)
+        {
+            var linkedNotification = new Notification();
+            context.Entry(linkedNotification.PatientDetails).CurrentValues.SetValues(notification.PatientDetails);
+
+            notification.LinkedNotifications.Add(new NotificationLink { LinkedNotification = linkedNotification });
+            linkedNotification.LinkedNotifications.Add(new NotificationLink { LinkedNotification = notification });
```

Comment:
This is the problem we'll have with the double-links approach - if one of these was already linked, that wouldn't have transitioned through to the new notification.